src/storage/txn/process.rs

```diff
@@ -109,32 +105,38 @@ impl Task {
     }
 }
 
-pub struct Executor<E: Engine> {
+pub trait MsgScheduler: Clone + Send + 'static {
+    fn on_msg(&self, task: Msg) -> ::std::result::Result<(), ScheduleError<Msg>>;
+}
+
+pub struct Executor<E: Engine, S: MsgScheduler> {
     // We put time consuming tasks to the thread pool.
     pool: Option<threadpool::Scheduler<SchedContext<E>>>,
     // And the tasks completes we post a completion to the `Scheduler`.
-    scheduler: Option<worker::Scheduler<Msg>>,
+    scheduler: Option<S>,
 }
 
-impl<E: Engine> Executor<E> {
-    pub fn new(
-        scheduler: worker::Scheduler<Msg>,
-        pool: threadpool::Scheduler<SchedContext<E>>,
-    ) -> Self {
+impl<E: Engine, S: MsgScheduler> Executor<E, S> {
+    pub fn new(scheduler: S, pool: threadpool::Scheduler<SchedContext<E>>) -> Self {
         Executor {
             scheduler: Some(scheduler),
             pool: Some(pool),
         }
     }
 
-    fn take_scheduler(&mut self) -> worker::Scheduler<Msg> {
+    fn take_scheduler(&mut self) -> S {
         self.scheduler.take().unwrap()
     }
 
     fn take_pool(&mut self) -> threadpool::Scheduler<SchedContext<E>> {
         self.pool.take().unwrap()
     }
 
+    #[inline]
```

Comment:
can remove inline here

Reply:
Done. Maybe we can remove all explict `inline` in our repos?